roles/openshift_health_checker/action_plugins/openshift_health_check.py

```diff
@@ -25,33 +25,37 @@ class ActionModule(ActionBase):
 
     def run(self, tmp=None, task_vars=None):
         result = super(ActionModule, self).run(tmp, task_vars)
-        task_vars = task_vars or {}
-
-        # vars are not supportably available in the callback plugin,
-        # so record any it will need in the result.
-        result['playbook_context'] = task_vars.get('r_openshift_health_checker_playbook_context')
-
-        if "openshift" not in task_vars:
-            result["failed"] = True
-            result["msg"] = "'openshift' is undefined, did 'openshift_facts' run?"
-            return result
 
         try:
             known_checks = self.load_known_checks()
+            args = self._task.args
+            requested_checks = normalize(args.get('checks', []))
+
+            if not requested_checks:
```

Comment:
Yes, they would fail if either `checks` is not passed or is passed and is empty.

Did I overlook some genuine case when we want to call the action plugin without arguments?

Note that it fails because that's the easiest way to cause text to go to stdout/stderr (without messing with the verbosity level).